### PR TITLE
Set entity from api adding transactionality

### PR DESCRIPTION
**Pull request by alexsilaghi**

### CODE REVIEW

File: src/main/java/edu/stanford/protege/webprotege/postcoordinationservice/repositories/PostCoordinationRepository.java [hunk 1]
```diff
@@ -132,4 +134,74 @@ public Optional<EntityCustomScalesValuesHistory> getExistingCustomScaleHistoryOr
...
+
+        Query query = new Query();
+        query.addCriteria(
+                Criteria.where(WHOFIC_ENTITY_IRI).is(entityIri)
+                        .and(PROJECT_ID).is(projectId.value())
+        );
+        Update update = new Update().pull("postCoordinationRevisions",
+                new Document("changeRequestId._id", changeRequestId.id()));
+
+        readWriteLock.executeReadLock(() -> {
```
Comment (soimugeoWB) [inline src/main/java/edu/stanford/protege/webprotege/postcoordinationservice/repositories/PostCoordinationRepository.java:163]: [question]: shouldn't this be a write lock?

File: src/main/java/edu/stanford/protege/webprotege/postcoordinationservice/repositories/PostCoordinationRepository.java [hunk 2]
```diff
@@ -132,4 +134,74 @@ public Optional<EntityCustomScalesValuesHistory> getExistingCustomScaleHistoryOr
...
+
+    public void deletePostCoordinationCustomScalesRevision(ChangeRequestId changeRequestId, ProjectId projectId, String entityIri) {
+
+        Query query = new Query();
+        query.addCriteria(
+                Criteria.where(WHOFIC_ENTITY_IRI).is(entityIri)
+                        .and(PROJECT_ID).is(projectId.value())
+        );
+        Update update = new Update().pull("postCoordinationCustomScalesRevisions",
+                new Document("changeRequestId._id", changeRequestId.id()));
```
Comment (soimugeoWB) [inline src/main/java/edu/stanford/protege/webprotege/postcoordinationservice/repositories/PostCoordinationRepository.java:146]: [suggestion]: in database i think we should only store the actual changeRequestId and not the ChangeRequestId object.

File: src/main/java/edu/stanford/protege/webprotege/postcoordinationservice/repositories/PostCoordinationRepository.java [hunk 3]
```diff
@@ -33,10 +36,8 @@ public PostCoordinationRepository(MongoTemplate mongoTemplate, ReadWriteLockServ
     }
 
     public void bulkWriteDocuments(List<InsertOneModel<Document>> listOfInsertOneModelDocument, String collectionName) {
-        readWriteLock.executeWriteLock(() -> {
```
[the remark this answers is not in the document]
Comment (alexsilaghi) [reply src/main/java/edu/stanford/protege/webprotege/postcoordinationservice/repositories/PostCoordinationRepository.java:36]: I added the lock back. I don't know how it got removed.

File: src/main/java/edu/stanford/protege/webprotege/postcoordinationservice/repositories/PostCoordinationRepository.java (hunk 1 above)
Comment (alexsilaghi) [reply src/main/java/edu/stanford/protege/webprotege/postcoordinationservice/repositories/PostCoordinationRepository.java]: changed to write lock.

File: src/main/java/edu/stanford/protege/webprotege/postcoordinationservice/repositories/PostCoordinationRepository.java (hunk 2 above)
Comment (alexsilaghi) [reply src/main/java/edu/stanford/protege/webprotege/postcoordinationservice/repositories/PostCoordinationRepository.java]: changed to string only.

**Pull request merged**